demos/common/cpp/utils/src/images_capture.cpp

```diff
@@ -190,6 +204,8 @@ class CameraCapWrapper : public ImagesCapture {
     double fps() const override {return cap.get(cv::CAP_PROP_FPS) > 0 ? cap.get(cv::CAP_PROP_FPS) : 30;}
 
     cv::Mat read() override {
+        auto startTime = std::chrono::steady_clock::now();
```

Comment:
I believe it's natural to say that the pipeline of processing and inference of an image starts with reading. That means start of a pipeline for a given image and it's reading is the same time point, but here you have two: one is here and one is in `main()`. You need to come up with better design to introduce reading latency metric.

Reply:
This metric measure only reading latency. So, generally speaking, start time for it is not the same as pipeline's (there might be some other preparation routines before reading in each iteration). 
But more important cause why start time is taken here is to measure reading latency inside the images_capture classes to avoid extra code / passing extra values from application level.

Reply:
> avoid extra code / passing extra values from application level

This can be done in `main()`, no need to pass anything.

> So, generally speaking, start time for it is not the same as pipeline's (there might be some other preparation routines before reading in each iteration).

There aren't such preparations. Do you have a specific example? Any way it is controlled by `main()`.

Reply:
If we will implement everything in main, we'll have to copy this code to all demos. Here it is implemented in one place.
Frankly I don't see a problem in taking 2 different start times - we measure different values, so they have different start times. As I said, frame reading might not be in the beginning of pipeline in some demos. For instance, we might check for processed frames first and render them and only after that read the new image.

Reply:
> If we will implement everything in main, we'll have to copy this code to all demos. Here it is implemented in one place.

That is why I have been claiming that the whole main loop should be a separate function, but not only what went into AsyncPipeline. BTW https://github.com/openvinotoolkit/open_model_zoo/pull/2214 wouldn't suffer than as well.

> Frankly I don't see a problem in taking 2 different start times

This is not a big problem, but I perceive this metric modifications only as your initiative, which means there is no rush, so we have time to do things right.

> For instance, we might check for processed frames first and render them and only after that read the new image.

We followed the idea minimizing latency so far because there are other ways to maximize throughput through cmd. Why would we change that?